app/src/main/java/net/gsantner/markor/util/BackupRestoreHelper.java

```diff
@@ -0,0 +1,203 @@
+            ".*\\.xml"
+    };
+
+    private static boolean includeFile(final File f) {
+        return includeFile(f.getPath());
+    }
+
+    private static boolean includeFile(final String string) {
+
+        boolean excluded = false;
+        for (final String pattern : EXCLUDE_PATTERNS) {
+            if (matchPattern(string, pattern)) {
+                excluded = true;
+                break;
+            }
+        }
+
+        boolean included = false;
+        for (final String pattern : INCLUDE_PATTERNS) {
+            if (matchPattern(string, pattern)) {
+                included = true;
+                break;
+            }
+        }
+
+        return included && !excluded;
+    }
+
+    private static boolean matchPattern(final String string, final String pattern) {
+        final boolean isRegex = pattern.contains("*");
+        return ((isRegex && string.matches(pattern)) || (!isRegex && string.contains(pattern)));
+    }
+
+    private static File getPrefDir(final Context context) throws PackageManager.NameNotFoundException {
+        final PackageManager packageManager = context.getPackageManager();
+        final PackageInfo p = packageManager.getPackageInfo(context.getPackageName(), 0);
+        return new File(p.applicationInfo.dataDir, "shared_prefs");
+    }
+
+    public static void backupConfig(final Context context, final FragmentManager manager) {
+
+        if (context instanceof Activity) {
+            final Activity activity = (Activity) context;
+
+            FilesystemViewerCreator.showFolderDialog(
+                    new FilesystemViewerData.SelectionListenerAdapter() {
+                        @Override
+                        public void onFsViewerConfig(FilesystemViewerData.Options dopt) {
+                            dopt.rootFolder = new AppSettings(context).getNotebookDirectory();
+                            dopt.titleText = R.string.select_folder;
+                        }
+
+                        @Override
+                        public void onFsViewerSelected(String request, File dir) {
+                            createAndSaveBackup(context, getSaveFile(dir));
+                        }
+                    }, manager, activity
+            );
+        }
+    }
+
+    public static File getSaveFile(final File folder) {
+        File file;
+        int index = 0;
+        do {
+            file = new File(folder, String.format(SAVE_NAME, (index == 0) ? "" : String.format("_%d", index)));
+            index++;
+        } while (file.exists());
+        return file;
+    }
+
+    public static void createAndSaveBackup(final Context context, final File saveLoc) {
+        try {
+            final FileOutputStream fos = new FileOutputStream(saveLoc);
+            final BufferedOutputStream bos = new BufferedOutputStream(fos);
+            final ZipOutputStream zos = new ZipOutputStream(bos);
+            final File[] files = getPrefDir(context).listFiles();
```

Comment:
Why not simply use _preferences.getAll() instead of reading file?

Reply:
I'm just copying / exporting the shared preference XML files as is. Not reading them or parsing their contents.

Reply:
Are you sure that this is an official api? What happens if a future android version will change the file format?
What if the file access will change again? I would recommend to only use official documented api.

Reply:
1. If Android breaks the File api, much of markor will be broken. I am comfortable using this.
2. `ApplicationInfo` / `dataDir` appear to be documented
3. As pointed out above, much of this logic is lifted from OpenLauncher.

Reply:
I'm 99% sure internal File access will stay, for private app store, so including prefs.

For external storage it might be different.